includes/widgets/traits/button-trait.php

```diff
@@ -503,6 +438,160 @@ protected function register_button_style_controls( $args = [] ) {
 		);
 	}
 
+	protected function register_icon_style_controls( $args = [] ) {
+		$default_args = [
+			'section_condition' => [],
+			'alignment_default' => '',
+			'alignment_control_prefix_class' => 'elementor%s-align-',
+			'content_alignment_default' => '',
+		];
+
+		$args = wp_parse_args( $args, $default_args );
+
+		$start = is_rtl() ? 'right' : 'left';
+		$end = is_rtl() ? 'left' : 'right';
+
+		$this->add_control(
+			'icon_align',
+			[
+				'label' => esc_html__( 'Position', 'elementor' ),
+				'type' => Controls_Manager::CHOOSE,
+				'options' => [
+					'row' => [
+						'title' => esc_html__( 'Start', 'elementor' ),
+						'icon' => "eicon-h-align-{$start}",
+					],
+					'row-reverse' => [
+						'title' => esc_html__( 'End', 'elementor' ),
+						'icon' => "eicon-h-align-{$end}",
+					],
+					'column' => [
+						'title' => esc_html__( 'Above', 'elementor' ),
+						'icon' => "eicon-v-align-top",
+					],
+					'column-reverse' => [
+						'title' => esc_html__( 'Below', 'elementor' ),
+						'icon' => "eicon-v-align-bottom",
+					],
+				],
+				'selectors_dictionary' => [
+					'row' => is_rtl() ? '--button-flex-direction: row-reverse' : '--button-flex-direction: row',
+					'row-reverse' => is_rtl() ? '--button-flex-direction: row' : '--button-flex-direction: row-reverse',
+					'column' => '--button-flex-direction: column; --button-align-self: center; --button-icon-display: flex;',
+					'column-reverse' => '--button-flex-direction: column-reverse; --button-align-self: center; --button-icon-display: flex;',
+				],
+				'selectors' => [
+					'{{WRAPPER}}' => '{{VALUE}};',
+				],
+				'condition' => array_merge(
+					$args['section_condition'],
+					[
+						'text!' => '',
+						'selected_icon[value]!' => '',
+					]
+				),
+			]
+		);
+
+		$this->add_responsive_control(
+			'icon_size',
+			[
+				'label' => esc_html__( 'Size', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'range' => [
+					'px' => [
+						'max' => 100,
+					],
+					'em' => [
+						'max' => 10,
+					],
+					'rem' => [
+						'max' => 10,
+					],
+				],
+				'size_units' => [ 'px', 'em', 'rem', 'vw', 'custom' ],
+				'selectors' => [
+					'{{WRAPPER}}' => '--button-icon-size: {{SIZE}}{{UNIT}}; --button-icon-display: flex; --button-align-self: center;',
```

Comment:
I have added the new` display` and` align self` styling here.
If the user leaves the icon size control empty, then we will keep the original styling.